### DISCUSSION TITLE
Add A callback function option in ''useIntersection'' hook

### DISCUSSION THREAD

**Discussion by khalids01** [mantinedev/mantine#2722] [Feature requests]
A callback function option is needed to perform some actions when the Element is intersecting.

**Comment by rtivital** [comment]
use-intersection hooks provides an api to work with [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API), it does not include any callbacks. You can subscribe to changes with `useEffect`.
